### ISSUE TITLE
WinForms IBindingList Collection Support

### ISSUE THREAD

**Issue by bchavez** [RolandPheasant/DynamicData#182]
Hello,

**ReactiveUI** has deprecated [`ReactiveBindingList<T>`](https://github.com/reactiveui/ReactiveUI/blob/4a7b6c7cb35efe4e936b6779e2ad60964894e1d0/src/ReactiveUI.Winforms/ReactiveBindingList.cs#L15) which had support for `IBindingList`. The obsolete message reads:

```
ReactiveList is no longer supported. We suggest replacing
it with DynamicData https://github.com/rolandpheasant/dynamicdata
```
So I searched here for any type that implements `IBindingList` and couldn't find anything. Effectively, without some type implementing `IBindingList` there doesn't seem to be a story for binding collections to **WinForms** controls such as a `ListBox`.

I think what we need is a type similar to **WPF**'s `ObservableCollectionExtended<T>`, but for **WinForms** like `ObservableBindingListExtended<T>` that implements `IBindingList` (or `BindingList<T>`) that can be used similar to:
```
this.SourceCache
   .Connect()
   .Bind(this.ObservableBindingListExtended)
   .Subscribe();
```

Let me know your thoughts.

Thanks,
Brian

**Comment by glennawatson** [comment]
@RolandPheasant You should actually be able to make a `ReadOnlyBindableCollection<T>` style container and use the Bind(out myList); type syntax.

**Comment by RolandPheasant** [comment]
Hi, I agree this needs adding to the lib. 

In the meantime there is a sample implementation which you can copy into your project [binding list example](https://github.com/RolandPheasant/DynamicData.Winforms.VirtualList/tree/master/DynamicDataEx)

**Comment by RolandPheasant** [comment]
The example in the link is for a list, not a cache.  I will ensure both list and cache are supported. Also I will post as cache example later so you can test it before release.

**Comment by RolandPheasant** [comment]
I have added `Bind()` support for binding list for both cache and list parts of the system.

What I have not done is to support `ToObservableChangeSet()` from BindingList.

Also the suggestion by @glennawatson to create a read-only binding list and bind using an out param is not implemented.  I will probably do it in the near future.

@bchavez since this is new functionality I could deploy to Nuget without breaking any code so I will do this very soon. However it would be good if you can download the packages from here [AppVeyor Build v99.99.99.2523-branches](https://ci.appveyor.com/project/RolandPheasant/dynamicdata-dpbpa) and test first

**Comment by RolandPheasant** [comment]
@bchavez  I've merge it in, so you need to download v6.7.0 from the AppVeyor artefacts

**Comment by bchavez** [comment]
Hey @RolandPheasant ,

I think it works!

![screen_2368](https://user-images.githubusercontent.com/478118/48377346-6c2fa980-e682-11e8-9b03-99ea2cae9d59.gif)

Thanks again so much for your help @RolandPheasant and @glennawatson . Such amazing work! I really appreciate it.

BTW, here is my `ViewModel` code in case anyone is interested:
```csharp
public class Account
{
   public int Id { get; set; }
   public string Name { get; set; }
   public string Number { get; set; }
}

public class MainViewModel : ReactiveObject
{
   public MainViewModel()
   {
      this.Accounts = new BindingList<Account>();
      this.AccountCache = new SourceCache<Account, int>(f => f.Id);


      this.AccountCache.Connect()
         .Bind(this.Accounts)
         .Subscribe();

      var hasAccounts = this.AccountCache.CountChanged
         .Select(x => x > 0);
      
      this.AddAccount = ReactiveCommand2.Create(AddAccountImpl, Observable.Return(true));
      this.RemoveAccount = ReactiveCommand2.Create(RemoveAccountImpl, hasAccounts);
   }

   private Account selectedAccount;
   public Account SelectedAccount
   {
      get => selectedAccount;
      set => this.RaiseAndSetIfChanged(ref selectedAccount, value);
   }

   public ReactiveCommand2<Unit, Unit> AddAccount { get; }
   public ReactiveCommand2<Unit, Unit> RemoveAccount { get; }

   public BindingList<Account> Accounts { get; }

   public SourceCache<Account, int> AccountCache { get; }

   private void RemoveAccountImpl()
   {
      if( this.selectedAccount != null )
         this.AccountCache.Remove(this.selectedAccount);
   }

   private int nextId = 0;

   private void AddAccountImpl()
   {
      var id = nextId++;
      var a = new Account
         {
            Id = id,
            Name = $"Account {id}",
            Number = "12345"
         };

      this.AccountCache.AddOrUpdate(a);
   }
}
```

I hope I'm using the new `IBindingList` feature correctly. I'm very noobish with ReactiveUI; I'm still learning! :cry:

**Comment by RolandPheasant** [comment]
That's great to see. I will deploy it now. 

Are you on slack? There is an active community with channels for Rx, RxUI and DynamicData. It is a great place to ask questions [Link to join slack](https://reactiveui.net/slack)

**Comment by RolandPheasant** [comment]
On second thoughts it is late here and I need to update the release notes before I deploy. Will do it in the next couple days.

**Comment by bchavez** [comment]
@RolandPheasant , yep! Just joined slack yesterday my username is @bchavez. It was @glennawatson on slack who suggested I raise the issue about `IBindingList` here. :)

Again, thanks for all the help and support! :heart: I have a long road ahead to learn all the ins and outs, but this `IBindingList` was a major blocker that's now been unblocked. :)

**Comment by RolandPheasant** [comment]
Feel free to ask any questions and report any issues

**Comment by RolandPheasant** [comment]
Fixed by PR #183
